hotsos/core/ycheck/events.py

```diff
@@ -225,7 +239,238 @@ def __call__(self):
+    # before loading a new set of event searches.
+    search_registry = EventsSearchRegistry()
+
+    @staticmethod
+    def meets_requirements(event):
+        """
+        If an event or group has a requirements property it must return True
+        for the events to be executed.
+        """
+        if HotSOSConfig.force_mode:
+            return True
+
+        if event.requires and not event.requires.result:
+            log.debug("event '%s' pre-requisites not met - "
+                      "skipping", event.name)
+            return False
+
+        return True
+
+    @staticmethod
+    def skip_filtered_event(event_path):
+        """
+        Apply event filter if provided.
+        """
+        e_filter = HotSOSConfig.event_filter
+        if e_filter and event_path != e_filter:
+            log.info("skipping event %s (filter=%s)", event_path, e_filter)
+            return True
+
+        return False
+
+    @staticmethod
+    def get_defs(group=None):
+        """
+        Load the event definitions for the current plugin. By default all are
+        loaded and if a group path is provided, only events that are part of
+        that path are included.
+        """
+        log.debug("loading event defs (group=%s)", group)
+        plugin_defs = YDefsLoader('events').plugin_defs
+        if not plugin_defs:
+            return {}
+
+        if not group:
+            return plugin_defs
+
+        # Exclude events that are not part of the group.
+        groups = group.split('.')
+        for i, _group in enumerate(groups):
+            if i == 0:
+                plugin_defs = {_group: plugin_defs[_group]}
+            else:
+                prev = groups[i - 1]
+                plugin_defs[prev] = {_group: plugin_defs[prev][_group]}
+
+        return plugin_defs
+
+    @classmethod
+    def load_searches(cls, group=None, searcher=None):
+        """
+        Find all events that have a search property and load their search into
+        the global searcher unless a custom searcher is provided here.
+        """
+        if searcher is None:
+            searcher = cls.search_registry.get_global_searcher()
+            if len(searcher.catalog) > 0:
+                raise Exception("global event searcher catalog is not empty "
+                                "and must be reset before loading so as not "
+                                "to include searches from a previous run.")
+
+        log.debug("started loading event (group=%s) searches into searcher "
+                  "(%s)", group, searcher)
+
+        search_props = set()
+        events = YDefsSection(HotSOSConfig.plugin_name,
+                              cls.get_defs(group) or {})
+        for prop in events.manager.properties.values():
+            for item in prop:
+                if not issubclass(item['cls'], search.YPropertySearch):
+                    break
+
+                search_props.add(item['path'])
+
+        if len(search_props) > 0:
```

Comment:
This condition could be reversed in order to have less indentation further down, i.e.:

```python

if not search_props:
    log.debug("no events found - no searches to load")
    return
```

Reply:
ack